sklearn/decomposition/kernel_pca.py

```diff
@@ -258,8 +278,10 @@ def fit(self, X, y=None):
         self._fit_transform(K)
 
         if self.fit_inverse_transform:
-            sqrt_lambdas = np.diag(np.sqrt(self.lambdas_))
-            X_transformed = np.dot(self.alphas_, sqrt_lambdas)
+            # Transform X
+            # (shortcut since we transform the same X that was used to fit)
+            X_transformed = self.alphas_ * np.sqrt(self.lambdas_)
```

Comment:
This line is pretty self explanatory, the comment above is not needed.

Reply:
I have doubts about this.. is there any danger in having too much explicit comments in the code ? If I were to read this code again two years from now I would be happy to see that :)

Reply:
The variable is named `X_transformed` and one comment is `# Transform X`. This seems a little redundant. 

`# (shortcut since we transform the same X that was used to fit)` I can see how this comment could be relevant in `fit_transform`. If one was reading this comment in two years, they will see that X is transformed, from the code and the comment. Not much more information is gained from the comment. The code has a little more information, since it tells you *how* it is being transformed.

Reply:
> The variable is named X_transformed and one comment is # Transform X. This seems a little redundant.

granted :)

My point was that in kernel PCA the "standard" way to transform is 

```python
np.dot(K, self.alphas_ / np.sqrt(self.lambdas_))
```

but here we use a shortcut formula that is only valid because we know that `X` is precisely the one used to learn the transform. 

So I propose this formulation 

```python
# Transform X: no need to recompute kernel, use shortcut
X_transformed = self.alphas_ * np.sqrt(self.lambdas_)
```